spec/mongoid/clients_spec.rb

```diff
@@ -731,27 +731,42 @@
 
     context "when the override is global" do
 
-      before do
-        Mongoid.override_database(:mongoid_optional)
-      end
+      shared_examples_for "a global database override" do
 
-      after do
-        Band.delete_all
-        Mongoid.override_database(nil)
-      end
+        before do
+          Mongoid.override_database(:mongoid_optional)
+        end
 
-      let!(:band) do
-        Band.create(name: "Tool")
-      end
+        after do
+          Band.delete_all
+          Mongoid.override_database(nil)
+        end
+
+        let!(:band) do
+          klass.create(name: "Tool")
+        end
+
+        it "persists to the overridden database" do
+          mongo_client = Band.mongo_client.with(database: :mongoid_optional)
+          expect(mongo_client[:bands].count(name: "Tool")).to eq(1)
+        end
 
-      it "persists to the overridden database" do
-        Band.mongo_client.with(database: :mongoid_optional) do |sess|
-          expect(sess[:bands].find(name: "Tool")).to_not be_nil
```

Comment:
This test was not executed. `with` does not take a block as argument, and I guess by the name `sess` that it was an old test that used to work with `moped`, but did not work anymore. It just did nothing at all.

Also, `find` would always return a `Mongo::Collection::View`, which is not `nil`, so the test would never fail!